File: majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int major = nums[0] , freq = 1;
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i] == major) freq++;
            else 
            {
                freq--;
                if(freq == 0) {freq = 1 ; major = nums[i];}
            }
        }
        
        return major;
    }
}

File: majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cand = nums[0] , cnt = 0;
        
        for(auto &ele : nums)
        {
            if(ele == cand) cnt++;
            else cnt--;
            
            if(cnt == 0) cand = ele , cnt = 1;
        }
        
        cnt = 0;
        
        for(auto &ele : nums)
        {
            cnt += (cand == ele);
        }
        
        return cnt > size(nums)/2 ? cand : -1;
        
    }
};